### ISSUE TITLE
Empty data when the client does not enable the pg simple protocol

### ISSUE THREAD

**Issue by neverchanje** [risingwavelabs/risingwave#7599]
### To Reproduce

```go
package main

import (
	"context"
	"fmt"
	"os"
	"time"

	"github.com/jackc/pgx/v5"
)

type RowItem struct {
	Id       string
	Distance float64
	Duration float64
}

func test() error {
	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
	defer cancel()

	// setup connection
	url := "postgresql://root@localhost:4566/dev"
	cfg, err := pgx.ParseConfig(url)
	if err != nil {
		return err
	}
	cfg.DefaultQueryExecMode = pgx.QueryExecModeSimpleProtocol
	conn, err := pgx.ConnectConfig(ctx, cfg)
	if err != nil {
		return fmt.Errorf("failed to open connection: %s", err.Error())
	}
	defer conn.Close(ctx)

	// ping db
	err = conn.Ping(ctx)
	if err != nil {
		return fmt.Errorf("failed to ping db: %s", err.Error())
	}

	rows, err := conn.Query(ctx, "SELECT * FROM taxi_trips;")
	if err != nil {
		return fmt.Errorf("failed to run SELECT statement: %s", err.Error())
	}
	defer rows.Close()
	for rows.Next() {
		var item RowItem
		rows.Scan(&item.Id, &item.Distance, &item.Duration)
		fmt.Println(item)
	}
	return nil
}

func main() {
	if err := test(); err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
}
```

```sql
CREATE TABLE taxi_trips(id STRING, distance DOUBLE, duration DOUBLE);

INSERT INTO taxi_trips VALUES ('1', 4, 4);
INSERT INTO taxi_trips VALUES ('1', 4, 4);
INSERT INTO taxi_trips VALUES ('1', 4, 4);
```

<img width="813" alt="Screenshot 2023-01-31 at 09 26 01" src="https://user-images.githubusercontent.com/6970676/215638163-e9eec03f-d7dc-4b04-bed7-eda3f1c1ed57.png">

If QueryExecModeSimpleProtocol is disabled, the result set will be empty.

### Additional context

commit: 198ca46c3b492e04cdbdef32d8d1a7c55dedfb74

**Comment by BowenXiao1999** [comment]
I remembered that we have discussed this problem before in slack:https://risingwave-labs.slack.com/archives/C034U0NH5ND/p1670231523906229 , it's relate to extended mode.


This one's problem is pg client in Go can not scan results that psql have inserted. Maybe there are some fundamental bugs. cc @ZENOTME , can you take a look? Thanks!

**Comment by ZENOTME** [comment]
Go will use different format code for different type, e.g. `STRING,DOUBLE -> TEXT,BINARY `

But now we only support uniform type.🥵 I will take a look to implement this mix format code feature.

But I think the problem in this case is that we didn't report the error to the user. I investigate and find that this error return from `pg_proto.read_message().await?;`, the error will not caught by `pg_proto.process` so that it will not be reported to user. 
```
            let msg = pg_proto.read_message().await?;
            tracing::trace!("Received message: {:?}", msg);
            let ret = pg_proto.process(msg).await;
```

One easy fix way is to add report here like: 
```
            let msg = pg_proto.read_message().await;
            match msg {
                Err(err) => //report err
            }
            tracing::trace!("Received message: {:?}", msg);
            let ret = pg_proto.process(msg).await;
```

@BowenXiao1999

**Comment by ZENOTME** [comment]
"uniform type" means uniform format, we only support to send the row as binary format or text format. We can't send the row with binary format for one column and text format for another column. That's mix format.

> Btw, the key is that we may need test workflows for SDKs in popular languages.
After #7622, we can support mix format so that we can support go pq.

**Comment by ZENOTME** [comment]
Closed by #7622
